metaflow/plugins/tag_cli.py

```diff
@@ -0,0 +1,531 @@
+        obj.echo_always(
+            " ; ".join(
+                [
+                    "system:%s" % group if is_system else "user:%s" % group,
+                    ",".join(runs),
+                ]
+            )
+        )
+
+
+def _print_tags_for_one_run(obj, run):
+    system_tags = {run.pathspec: run.system_tags}
+    all_tags = {run.pathspec: run.tags}
+    return _print_tags_for_runs_by_groups(obj, system_tags, all_tags, by_tag=False)
+
+
+def _is_prod_token(token):
+    return token and token.startswith("production:")
+
+
+def _get_client_run_obj(obj, run_id, user_namespace):
+    if _is_prod_token(user_namespace):
+        raise CommandException(
+            "Modifying a scheduled run's tags is currently not allowed."
+        )
+
+    flow_name = obj.flow.name
+
+    # handle error messaging for two cases
+    # 1. our user tries to tag a new flow before it is run
+    # 2. our user makes a typo in --namespace
+    try:
+        namespace(user_namespace)
+        Flow(pathspec=flow_name)
+    except MetaflowNotFound:
+        raise CommandException(
+            "No execution found for *%s*. Please run the flow before tagging."
+            % flow_name
+        )
+
+    except MetaflowNamespaceMismatch:
+        raise CommandException(
+            "No execution found for *%s* in namespace *%s*."
+            % (flow_name, user_namespace)
+        )
+
+    # throw an error with message to include latest run-id when run_id is None
+    if run_id is None:
+        latest_run_id = Flow(pathspec=flow_name).latest_run.id
+        msg = (
+            "Please specify a run-id using --run-id.\n"
+            "*%s*'s latest execution in namespace *%s* has id *%s*."
+            % (flow_name, user_namespace, latest_run_id)
+        )
+        raise CommandException(msg)
+    run_id_parts = run_id.split("/")
+    if len(run_id_parts) == 1:
+        run_name = "%s/%s" % (flow_name, run_id)
+    else:
+        raise CommandException("Run-id *%s* is not a valid run-id" % run_id)
+
+    # handle error messaging for three cases
+    # 1. our user makes a typo in --run-id
+    # 2. our user tries to mutate tags on a meson/production run
+    # 3. our user's --run-id does not exist in the default/specified namespace
+    try:
+        namespace(user_namespace)
+        run = Run(pathspec=run_name)
+    except MetaflowNotFound:
+        raise CommandException("No run *%s* found" % run_name)
+    except MetaflowNamespaceMismatch:
+        namespace(None)
+        run = Run(pathspec=run_name)
+
+        if any([_is_prod_token(tag) for tag in run.system_tags]):
+            raise CommandException(
+                "Modifying a scheduled run's tags is currently not allowed."
+            )
+
+        msg = "Run *%s* does not belong to namespace *%s*\n" % (
+            run_name,
+            user_namespace,
+        )
+
+        user_identifier_tags = [
+            tag for tag in run.system_tags if tag.startswith("user:")
```

Comment:
Not every run will have a "user:.." tag - all runs executed on top of SFN/Argo/Airflow for example. How about we drop this functionality altogether?

Reply:
It seems reasonable to drop - unless it is well-beloved by existing users (cc @romain-intel )